### DISCUSSION TITLE
Primary key that doesn't autoincrement.

### DISCUSSION THREAD

**Discussion by jonyscathe** [fastapi/sqlmodel#1120] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
class HeroTeam(HeroBase, TeamBase, table=True):
    id: int = Field(primary_key=True, sa_column_kwargs={'autoincrement': False})
```

### Description

If I have a situation where I have a primary key that is set by some other entity external to the database then I don't want it to autoincrement.
Is the only way to do this to use `sa_column_kwargs={'autoincrement': False}` as above?
It does work, but it is hardly the nicest method for controlling autoincrement...

### Operating System

Linux

### SQLModel Version

0.0.22

### Python Version

3.12.6

**Comment by YuriiMotov** [comment]
There is no other way for now.
We should use `sa_column_kwargs={'autoincrement': False}`
